plugins/uefi/fu-plugin-uefi.c

```diff
@@ -135,6 +137,202 @@ fu_plugin_get_results (FuPlugin *plugin, FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_plugin_uefi_update_resource (fwup_resource *re,
+				guint64 hardware_instance,
+				GBytes *blob,
+				GError **error)
+{
+	int rc;
+	rc = fwup_set_up_update_with_buf (re, hardware_instance,
+					  g_bytes_get_data (blob, NULL),
+					  g_bytes_get_size (blob));
+	if (rc < 0) {
+		g_autoptr(GString) str = g_string_new (NULL);
+		rc = 1;
+		for (int i = 0; rc > 0; i++) {
+			char *filename = NULL;
+			char *function = NULL;
+			char *message = NULL;
+			int line = 0;
+			int err = 0;
+
+			rc = efi_error_get (i, &filename, &function, &line,
+					    &message, &err);
+			if (rc <= 0)
+				break;
+			g_string_append_printf (str, "{error #%d} %s:%d %s(): %s: %s, ",
+						i, filename, line, function,
+						message, strerror (err));
+		}
+		if (str->len > 2)
+			g_string_truncate (str, str->len - 2);
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "UEFI firmware update failed: %s",
+			     str->str);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static GBytes *
+fu_plugin_uefi_get_splash_data (guint width, guint height, GError **error)
+{
+	const gchar * const *langs = g_get_language_names ();
+	const gchar *localedir = LOCALEDIR;
+	const gsize chunk_size = 1024 * 1024;
+	gsize buf_idx = 0;
+	gsize buf_sz = chunk_size;
+	gssize len;
+	g_autofree gchar *basename = NULL;
+	g_autofree guint8 *buf = NULL;
+	g_autoptr(GBytes) compressed_data = NULL;
+	g_autoptr(GConverter) conv = NULL;
+	g_autoptr(GInputStream) stream_compressed = NULL;
+	g_autoptr(GInputStream) stream_raw = NULL;
+
+	/* ensure this is sane */
+	if (!g_str_has_prefix (localedir, "/"))
+		localedir = "/usr/share/locale";
+
+	/* find the closest locale match */
+	basename = g_strdup_printf ("fwupd-%u-%u.bmp.gz", width, height);
+	for (guint i = 0; langs[i] != NULL; i++) {
+		g_autofree gchar *fn = NULL;
+		if (g_str_has_suffix (langs[i], ".UTF-8"))
+			continue;
+		fn = g_build_filename (localedir, langs[i],
+				       "LC_IMAGES", basename, NULL);
+		if (g_file_test (fn, G_FILE_TEST_EXISTS)) {
+			compressed_data = fu_common_get_contents_bytes (fn, error);
+			if (compressed_data == NULL)
+				return NULL;
+			break;
+		}
+		g_debug ("no %s found", fn);
+	}
+
+	/* we found nothing, even LANG=en and LANG=C */
```

Comment:
I think this comment's content (what is in langs) is more useful at the start of the for loop.  I wasn't aware that `g_get_language_names `returned en and C until I looked at https://developer.gnome.org/glib/stable/glib-I18N.html.

Reply:
Fixed